testing/e2e/agent_install_test.go

```diff
@@ -169,122 +150,107 @@ func (suite *AgentInstallSuite) extractZip(r io.Reader) {
-// Additionally the elastic-agent symlink will be recreated.
 func (suite *AgentInstallSuite) extractTar(r io.Reader) {
 	suite.T().Helper()
-	var fleetPath, agentSrc, agentDst, agentLink string
 	// Extract tar.gz stream
-	stream, err := gzip.NewReader(r)
+	gs, err := gzip.NewReader(r)
 	suite.Require().NoError(err)
-	tarReader := tar.NewReader(stream)
-	for header, err := tarReader.Next(); err == nil; header, err = tarReader.Next() {
-		if header.FileInfo().IsDir() {
-			// headers may be specified out of order in the archive, as a shortcut we'll make all nested dirs
-			err := os.MkdirAll(filepath.Join(suite.downloadPath, header.Name), 0755)
-			suite.Require().NoError(err)
-			continue
-		}
-		if !header.FileInfo().Mode().IsRegular() {
-			// the elastic-agent may count as a symlink in Linux or MacOS bundles
-			if header.FileInfo().Name() == agentName {
-				agentDst = filepath.Join(suite.downloadPath, header.Name)
-				agentLink = header.Linkname
-				continue
-			}
-			suite.T().Logf("unable to extract irregular file: %s linkname: %s", header.Name, header.Linkname)
-			continue
-		}
-
-		var pathErr *os.PathError
-		dst, err := os.Create(filepath.Join(suite.downloadPath, header.Name))
-		// if we get a PathError, try to make the directory before retrying file creation
-		// headers may be specified out of order in the archive, as a shortcut we'll make all nested dirs
-		if errors.As(err, &pathErr) {
-			dir := filepath.Dir(header.Name)
-			err = os.MkdirAll(filepath.Join(suite.downloadPath, dir), 0755)
-			suite.Require().NoErrorf(err, "unable to create directory %s", dir)
-			dst, err = os.Create(filepath.Join(suite.downloadPath, header.Name))
-			suite.Require().NoErrorf(err, "unable to create file %s", header.Name)
-		} else if err != nil {
-			suite.Require().Failf("unable to create file", "filename: %s, error: %v", header.Name, err)
+	tarReader := tar.NewReader(gs)
+	for {
+		header, err := tarReader.Next()
+		if errors.Is(err, io.EOF) {
+			break
 		}
-		err = dst.Chmod(0755)
-		suite.Require().NoError(err)
-		_, err = io.Copy(dst, tarReader)
-		dst.Close() // might be a dirty close
 		suite.Require().NoError(err)
 
-		// note the fleet-server locaction in the extracted path
-		if strings.HasSuffix(header.Name, binaryName) {
-			fleetPath = filepath.Join(suite.downloadPath, header.Name)
-		}
-		// note if elastic-agent has been extracted as a regular file
-		if strings.HasSuffix(header.Name, agentName) {
-			suite.agentPath = filepath.Join(suite.downloadPath, header.Name)
-			agentSrc = filepath.Join(suite.downloadPath, header.Name)
-		}
-		// note the source of the elastic-agent if it has been detected as a symlink
-		if agentLink != "" && strings.HasSuffix(header.Name, agentLink) {
-			agentSrc = filepath.Join(suite.downloadPath, header.Name)
+		path := filepath.Join(suite.downloadPath, header.Name)
+		mode := header.FileInfo().Mode()
+		switch {
+		case mode.IsDir():
+			err := os.MkdirAll(path, 0755)
```

Comment:
Maybe add an os.Chmod() to enforce the correct permissions on the directory ?

Reply:
Do you think that we need that for a test directory?